### PR TITLE
fix: Support uint64 in crash reports

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add support for uint64 when decoding crash reports.

## :bulb: Motivation and Context

Applied from https://github.com/kstenerud/KSCrash/pull/375. Thanks, @GLinnik21, for the improvement.

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by GLinnik21** [comment]
You're welcome)

**Review by armcknight — APPROVED** [review]
LGTM, just some questions and suggestions but no blockers!

**Comment by armcknight** [comment]
In case you're wondering about the force push from me, I meant to send a fixup commit to one of your other PRs but hadn't paid attention to a failed branch switch locally, so instead committed it to this branch 🙃 All is as it was before I started meddling now.

File: Tests/SentryTests/SentryCrash/SentryCrashJSONCodec_Tests.m [hunk 1]
```diff
@@ -126,55 +126,56 @@ - (void)testSerializeDeserializeArrayInteger
...
-    XCTAssertNil(error, @"");
-    XCTAssertEqualObjects(result, original, @"");
+    XCTAssertNotNil(result);
+    XCTAssertNil(error);
+    XCTAssertEqualObjects(result, original);
 }
 
 - (void)testSerializeDeserializeArrayFloat
 {
     NSError *error = (NSError *)self;
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Tests/SentryTests/SentryCrash/SentryCrashJSONCodec_Tests.m:145]: It doesn't make sense. It was like that for all the other test cases as well. I removed it.

File: Sources/SentryCrash/Recording/Tools/SentryCrashJSONCodec.c [hunk 2]
```diff
@@ -1209,9 +1210,12 @@ decodeElement(const char *const name, SentryCrashJSONDecodeContext *context)
             return SentryCrashJSON_ERROR_INCOMPLETE;
         }
 
-        if (!isFPChar(*context->bufferPtr) && accum >= 0) {
-            accum *= sign;
-            return context->callbacks->onIntegerElement(name, accum, context->userData);
+        if (!isFPChar(*context->bufferPtr) && !isOverflow) {
+            if (sign > 0 || accum <= ((uint64_t)LLONG_MAX + 1)) {
+                int64_t signedAccum = accum;
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/SentryCrash/Recording/Tools/SentryCrashJSONCodec.c:1215]: I changed the code a bit. We only have support for `onIntegerElement` at the moment. I don't want to add `onUIntegerElement` right now and keep using the fallback of double.

File: Tests/SentryTests/SentryCrash/SentryCrashJSONCodec_Tests.m [hunk 3]
```diff
@@ -126,55 +126,56 @@ - (void)testSerializeDeserializeArrayInteger
...
+    XCTAssertNil(error);
+    XCTAssertEqual([[result objectAtIndex:0] floatValue], value);
 }
 
 - (void)testSerializeDeserializeArrayFloat2
 {
     NSError *error = (NSError *)self;
     NSString *expected = @"[-2e-15]";
-    id original = [NSArray arrayWithObjects:[NSNumber numberWithFloat:-2e-15f], nil];
+    float value = -2e-15f;
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Tests/SentryTests/SentryCrash/SentryCrashJSONCodec_Tests.m:166]: It was already like that. I don't want to change this in this PR, I only simplified the code a bit.

**Pull request merged**